src/Autonomous.cpp

```diff
@@ -54,9 +86,40 @@ void Autonomous::autonomyIter() {
   //  output: movement command
   //
 
-  setCmdVel(0.5, 1.0);
+  if (arrived(gpsPose)) { //toPose(gps, currHeading)
+    std::cout << "arrived at gate" << std::endl;
+    std::cout << "x: " << gpsPose(0) << " y: " << gpsPose(1) << " theta: " << gpsPose(2) << std::endl;
+
+
+    // implement moving through gates
+    // then update target
+  } else {
+    double dtheta = pathDirection(lidar, gpsPose);
+    double speed = 10.0;
+    // TODO incredibly clever algorithms for state estimation
+    // and path planning and control!
+    
+    if(!lidar.empty()) {
+        speed = 5.0;
+    }
+    setCmdVel(dtheta, speed);
+  }
+}
+
+double Autonomous::pathDirection(points_t lidar, pose_t gpsPose) {
+    double dtheta;
+    if(lidar.empty()) {
+        dtheta = angleToTarget(gpsPose);
+    } else {
+        const std::vector<PointXY>& ref = points_tToPointXYs(lidar);
+        PointXY direction = pather.getPath(ref, (float) gpsPose(0), (float) gpsPose(1), target);
```

Comment:
Does `getPath` return a direction or a location? E.g. if the robot is at the point (1,1) and the next spot on the path is (2,1), then will `getPath` return (1,0) or (2,1)? If the latter, then we would need to subtract the current robot position from the return value of `getPath` in order to get a direction. But again, let's wait until a future diff to dig into the problems with the Pather2 class.